rossum_api/models/hook.py

```diff
@@ -84,21 +105,116 @@ class Hook:
     name: str
     url: str
     active: bool
-    config: dict[str, Any]
-    test: dict[str, Any]
+    config: JsonDict
+    test: JsonDict
     guide: str | None
     read_more_url: str | None
     extension_image_url: str | None
-    type: Literal["webhook", "function"] = "webhook"
-    metadata: dict[str, Any] = field(default_factory=dict)
+    type: HookType = "webhook"
+    metadata: JsonDict = field(default_factory=dict)
     queues: list[str] = field(default_factory=list)
     run_after: list[str] = field(default_factory=list)
-    events: list[str] = field(default_factory=list)
-    settings: dict[str, Any] = field(default_factory=dict)
-    settings_schema: dict[str, Any] | None = None
-    secrets: dict[str, Any] = field(default_factory=dict)
-    extension_source: str = "custom"
+    events: list[HookEvent] = field(default_factory=list)
+    settings: JsonDict = field(default_factory=dict)
+    settings_schema: JsonDict | None = None
+    secrets: JsonDict = field(default_factory=dict)
+    extension_source: HookExtensionSource = "custom"
     sideload: list[str] = field(default_factory=list)
     token_owner: str | None = None
     token_lifetime_s: int | None = None
     description: str | None = None
+
+
+@dataclass
+class HookRunData:
+    """Data class for hook execution logs.
+
+    HookRunData captures detailed execution logs and metadata for webhook/hook runs within the system.
+    It provides structured logging for tracking hook lifecycle events, performance metrics, and debugging information.
+
+    Notes
+    -----
+    The ``timestamp`` field is automatically set to the current UTC time if not provided.
```

Comment:
I don't think this is useful and understandable. It just raises questions like who can provide the timestamp? When? If it's not provided will I get the current `timestamp` of the time when I am getting the logs? Or?